index.js

```diff
@@ -27,26 +27,29 @@ module.exports = {
     }
   },
 
-  setupPreprocessorRegistry(type, registry) {
-    if (type === 'parent') {
-      this._assignOptions(this.app);
-
-      let pluginFnName = this._stripTestSelectors ? '_buildStripPlugin' : '_buildHashParamPlugin';
-
-      let plugin = this[pluginFnName]();
-      plugin.parallelBabel = {
-        requireFile: __filename,
-        buildUsing: pluginFnName,
-        params: {},
-      };
+  _setupPreprocessorRegistry(registry) {
+    let pluginFnName = this._stripTestSelectors ? '_buildStripPlugin' : '_buildHashParamPlugin';
+
+    let plugin = this[pluginFnName]();
+    plugin.parallelBabel = {
+      requireFile: __filename,
+      buildUsing: pluginFnName,
+      params: {},
+    };
 
-      registry.add('htmlbars-ast-plugin', plugin);
-    }
+    registry.add('htmlbars-ast-plugin', plugin);
   },
 
   included(app) {
     this._super.included.apply(this, arguments);
 
+    let host = this._findHost();
+    this._assignOptions(host);
+
+    // we can't use the setupPreprocessorRegistry() hook as it is called too
+    // early and we do not have reliable access to `app.tests` there yet
+    this._setupPreprocessorRegistry(app.registry);
```

Comment:
Its weird that this is called `app`, the thing passed in is only an `lib/broccoli/ember-app.js` instance when this addon is a top level project addon but when its a dep of another addon then the thing passed in is the parent (which is an `lib/models/addon.js` instance).

I'd suggest renaming to `appOrParent`...

```suggestion
    this._setupPreprocessorRegistry(appOrParent.registry);
```

Reply:
👌